### DISCUSSION TITLE
Vault feature for OIDC identity provider client secret

### DISCUSSION THREAD

**Discussion by baranpost** [keycloak/keycloak#28973] [Q&A]
Hello everyone,

Version 24.0.3

Trying to create an identity provider of typ OIDC and use the vault feature to store the client secret.

I created my own docker image making sure to add the attribute --vault=file to the build step, also added to the start stage --vault-dir=/my/path to the place where I created a file (realm_desiredname as per documentation) and then at the client secret field I typed in ${vault.realm_desiredname} also per documentation. It seems that keycloak won't pick up the client secret from the vault file. Any idea what am I doing wrong here? Any bug I'm not aware of?

thanks!

**Comment by codespearhead** [comment]
Did you make sure to follow the [underscores conventions](https://www.keycloak.org/server/vault#:~:text=Using%20underscores%20in%20the%20Name)?

I've never used vault, but check if [this applies to you](https://github.com/keycloak/keycloak/discussions/16446#discussioncomment-6687357).

Possible duplicate: #16446 .
